lib/ruby/all/gem.rbi

```diff
@@ -0,0 +1,4568 @@
+
+        def dependencies_for(specification); end
+
+        def name_for(dependency); end
+
+        def name_for_explicit_dependency_source; end
+
+        def name_for_locking_dependency_source; end
+
+        def requirement_satisfied_by?(requirement, activated, spec); end
+
+        def search_for(dependency); end
+
+        def sort_dependencies(dependencies, activated, conflicts); end
+      end
+
+      module UI
+        def after_resolution; end
+
+        def before_resolution; end
+
+        def debug(depth = T.unsafe(nil)); end
+
+        def debug?; end
+
+        def indicate_progress; end
+
+        def output; end
+
+        def progress_rate; end
+      end
+
+      class VersionConflict
+        def conflicts; end
+
+        def initialize(conflicts); end
+      end
+    end
+
+    class RequirementList
+      include Enumerable
+      Elem = type_member(fixed: T.untyped)
+
+      def add(req); end
+
+      def each(&blk); end
+
+      def empty?; end
+
+      def next5; end
+
+      def remove; end
+
+      def size; end
+    end
+
+    class Set
+      def errors; end
+
+      def errors=(errors); end
+
+      def find_all(req); end
+
+      def prefetch(reqs); end
+
+      def prerelease; end
+
+      def prerelease=(prerelease); end
+
+      def remote; end
+
+      def remote=(remote); end
+
+      def remote?; end
+    end
+
+    class SourceSet
+      def add_source_gem(name, source); end
+    end
+
+    class SpecSpecification
+      def initialize(set, spec, source = T.unsafe(nil)); end
+    end
+
+    class Specification
+      def dependencies; end
+
+      def download(options); end
+
+      def fetch_development_dependencies; end
+
+      def full_name; end
+
+      def install(options = T.unsafe(nil)); end
+
+      def installable_platform?; end
+
+      def local?; end
+
+      def name; end
+
+      def platform; end
+
+      def set; end
+
+      def source; end
+
+      def spec; end
+
+      def version; end
+    end
+
+    class Stats
+      PATTERN = T.let(nil, T.untyped)
+
+      def backtracking!; end
+
+      def display; end
+
+      def iteration!; end
+
+      def record_depth(stack); end
+
+      def record_requirements(reqs); end
+
+      def requirement!; end
+    end
+
+    class VendorSet
+      def add_vendor_gem(name, directory); end
+
+      sig do
+        params(
+          name: T.untyped,
+          version: T.untyped,
+          platform: T.untyped,
+          source: T.untyped
+        ).returns(T.untyped)
+      end
+
+      def load_spec(name, version, platform, source); end
+
+      def pretty_print(q); end
+
+      def specs; end
+    end
+
+    class VendorSpecification
+      def ==(other); end
+    end
+  end
+
+  class RuntimeRequirementNotMetError
+    def suggestion; end
+
+    def suggestion=(suggestion); end
+  end
+
+  module Security
+    AlmostNoSecurity = T.let(nil, T.untyped)
+    DIGEST_NAME = T.let(nil, T.untyped)
+    EXTENSIONS = T.let(nil, T.untyped)
+    H
```

Comment:
Did you mean to `extend Enumerable` here? That's likely what's causing problems with the bundler rbi.

Reply:
oh wow, good catch. I'm not entirely sure how that got in there.

Reply:
You should also be able to remove the changes to the bundler rbi now.

Reply:
👍 done